src/util/BoxSizing.hxx

```diff
@@ -0,0 +1,101 @@
+#ifndef BOX_SIZING_HXX
+#define BOX_SIZING_HXX
+
+#include "../util/Meta.hxx"
+
+/**
+ * @class       ScreenMeasurement
+ * @brief       Represents a measurement made on the screen
+ * @details     This class is used to encode linear combinations
+ *              of absolute and relative measurements
+ */
+struct ScreenMeasurement
+{
+  /** @brief    Number of pixels */
+  int px;
+  /** @brief    Percentage of an element's width */
+  int ew;
+  /** @brief    Percentage of an element's height */
+  int eh;
+  /** @brief    Percentage of a layout's width */
+  int lw;
+  /** @brief    Percentage of a layout's height */
+  int lh;
+  /** @returns  The measurement result in pixels */
+  int measure(const class ScreenMeasurementValuation &);
+};
+
+/* Literal operators */
+constexpr ScreenMeasurement operator""_lh(unsigned long long);
+constexpr ScreenMeasurement operator""_lw(unsigned long long);
+constexpr ScreenMeasurement operator""_eh(unsigned long long);
+constexpr ScreenMeasurement operator""_ew(unsigned long long);
+constexpr ScreenMeasurement operator""_px(unsigned long long);
+
+/* Arithmetic operators */
+constexpr ScreenMeasurement operator+(const ScreenMeasurement &, const ScreenMeasurement &);
+constexpr ScreenMeasurement operator-(const ScreenMeasurement &, const ScreenMeasurement &);
+
+/** 
+ * @class       ScreenMeasurementValuation
+ * @brief       Represents values of pixels to measure a ScreenMeasurement
```

Comment:
The comment is unclear, at least for me.

Reply:
i also don't understand it. what are values of pixel, what is a screenmeasurement? is this the difference between points in pixel? will this be different when the ui is zoomed?

Reply:
ScreenMeasurement represents a way for people to represent measurements in many ways. The classic way is by pixels. Then there is as a percentage of screen width/height, and also as a percentage of layout width/height. ScreenMeasurementValuation is a structure that holds the values of window width/height layout width/height to actually measure a ScreenMeasurement. This allows someone to say "This Button should take half of the layout's width!" and whenever the window's size changes, a new measurement is made.